### PR TITLE
Update annotations for data dictionary

### PR DESCRIPTION
**Pull request by nllong**
This PR addresses #273.

* Update the annotations of the measure technology categories.
* Upgrade dependencies
* Update generate_data_dictionary script to add in a nice display name for the tech category

### CODE REVIEW

File: BuildingSync.xsd [hunk 1]
```diff
@@ -8,7 +8,9 @@
       <p>This document details the deprecation policy for the BuildingSync Schema.</p>
       <h2>Deprecating schema elements</h2>
       <p>Since the development of the BuildingSync Schema is application-driven, the schema document itself has evolved over time to reflect the evolving understanding of the problem space.</p>
-      <p><b>Rule #1: Schema elements may only be removed by incrementing the major version of the schema document.</b></p>
+      <p>
```
Comment (markborkum) [inline BuildingSync.xsd:11]: Should this be in a separate pull request?

**Review by markborkum — CHANGES_REQUESTED** [review]
Changes look good, thanks.

Should the formatting change in the XSD file be part of this pull request?

File: BuildingSync.xsd (hunk 1 above)
Comment (nllong) [reply BuildingSync.xsd:11]: yeah, i'll make it a separate pr. thanks.

**Review by macintoshpie — APPROVED** [review]
Looks good to me 💯
